resources/js/components/Preferences.vue

```diff
@@ -173,81 +232,196 @@
+                    return sysPrefs;
+                }),
+                userPreferences: computed(_ => {
+                    const userPrefs = store.getters.preferences;
+                    return userPrefs;
+                }),
+                preferences: computed(_ => {
+                    if(state.category == 'system') {
+                        return state.systemPreferences;
+                    } else {
+                        return state.userPreferences;
+                    }
+                }),
+                pluginPreferences: computed(_ => store.getters.pluginPreferences),
+                categories: computed(_ => {
+                    const cats = {
+                        user: {},
+                        system: {},
+                    };
+                    for(let k in preferencesConfig.user) {
+                        const subcategory = preferencesConfig.user[k];
+                        cats.user[k] = {
+                            preferences: subcategory.preferences.slice(),
+                        };
+                    }
+                    for(let k in preferencesConfig.system) {
+                        const subcategory = preferencesConfig.system[k];
+                        cats.system[k] = {
+                            preferences: subcategory.preferences.slice(),
+                        };
+                    }
+                    for(let k in state.pluginPreferences.user) {
+                        const subcategory = state.pluginPreferences.user[k];
+                        if(!cats.user[k]) {
+                            cats.user[k] = {
+                                preferences: subcategory.preferences.slice(),
+                            };
+                            if(subcategory.custom) {
+                                cats.user[k].custom = true;
+                                cats.user[k].title = subcategory.title;
+                            }
+                        } else {
+                            cats.user[k].preferences = cats.user[k].preferences.concat(subcategory.preferences.slice());
+                        }
+                    }
+                    for(let k in state.pluginPreferences.system) {
+                        const subcategory = state.pluginPreferences.system[k];
+                        if(!cats.system[k]) {
+                            cats.system[k] = {
+                                preferences: subcategory.preferences.slice(),
+                            };
+                            if(subcategory.custom) {
+                                cats.system[k].custom = true;
+                                cats.system[k].title = subcategory.title;
+                            }
+                        } else {
+                            cats.system[k].preferences = cats.system[k].preferences.concat(subcategory.preferences.slice());
+                        }
+                    }
```

Comment:
This is 2 times the same 2 code blocks (DRY). I refactored this and will push the changes soon.

Reply:
Also I renamed cats to categories, as I was confused what cats 🐱 are doing in our application

Reply:
refactored in #[b638606](https://github.com/DH-Center-Tuebingen/Spacialist/pull/494/commits/b638606b4bf415c62f9561ab2f8854cffd68df87)

Reply:
I like 🐱
:sweat_smile: